### PR TITLE
fix: Avatar.Group item could be wrapped

### PR DESCRIPTION
**Pull request by afc163**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Avatar.Group item margin when item is wrapped by other elements.         |
| 🇨🇳 Chinese |   修复 Avatar.Group 内 Avatar 外层包裹其他元素时间距样式失效问题。        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/avatar/demo/group.tsx [hunk 1]
```diff
@@ -6,7 +6,9 @@ const App: React.FC = () => (
   <>
     <Avatar.Group>
       <Avatar src="https://joeschmoe.io/api/v1/random" />
-      <Avatar style={{ backgroundColor: '#f56a00' }}>K</Avatar>
+      <a href="https://ant.design">
+        <Avatar style={{ backgroundColor: '#f56a00' }}>K</Avatar>
+      </a>
```
Comment (afc163) [inline components/avatar/demo/group.tsx:11]: 这种情况，应该生效。

File: components/avatar/style/index.ts [hunk 2]
```diff
@@ -119,7 +119,7 @@ const genGroupStyle: GenerateStyle<AvatarToken> = (token) => {
       [`${componentCls}`]: {
         borderColor: avatarGroupBorderColor,
 
-        [`&:not(:first-child)`]: {
+        [`&:not(&-group-item-0)`]: {
```
Comment (MadCcc) [inline components/avatar/style/index.ts:122]: 这个对 Group 里第一个是 a 标签的不生效，class 是打在 a 标签上的

File: components/avatar/demo/group.tsx (hunk 1 above)
Comment (zombieJ) [reply components/avatar/demo/group.tsx:11]:
```tsx
const MyEmpty = () => null;

<Avatar.Group>
  <MyEmpty />
  <Avatar src="https://joeschmoe.io/api/v1/random" />
</Avatar.Group>
```

这个时候第一个不是实元素，item-0 不是第一个。

File: components/avatar/style/index.ts (hunk 2 above)
Comment (afc163) [reply components/avatar/style/index.ts:122]: 👍

File: components/avatar/style/index.ts [hunk 3]
```diff
@@ -162,8 +152,7 @@ export default genComponentStyleHook('Avatar', (token) => {
     avatarFontSizeBase: Math.round((fontSizeLG + fontSizeXL) / 2),
     avatarFontSizeLG: fontSizeHeading3,
     avatarFontSizeSM: fontSize,
-    avatarGroupOverlapping: marginXS,
-    avatarGroupSpace: marginXS,
+    avatarGroupSpace: -marginXS,
```
Comment (afc163) [inline components/avatar/style/index.ts:155]: 这两个变量重复了，合并掉。
另外默认应该就是负数。

File: components/avatar/style/index.ts [hunk 4]
```diff
@@ -110,24 +110,18 @@ const genBaseStyle: GenerateStyle<AvatarToken> = (token) => {
...
-        },
       },
 
-      [`&-popover`]: {
-        [`${componentCls} + ${componentCls}`]: {
-          marginInlineStart: avatarGroupSpace,
-        },
+      [`> *:not(:first-child)`]: {
+        marginInlineStart: avatarGroupSpace,
       },
```
Comment (afc163) [inline components/avatar/style/index.ts:125]: 换了最简单的方案。

**Review by MadCcc — APPROVED** [review]

**Pull request merged**